### ISSUE TITLE
~30% nodes were blacklisted after successfull PoW round

### ISSUE THREAD

**Issue by lehisnoe** [Zilliqa/Zilliqa#1797]
A portion of nodes (~20-30%) were blacklisted after succesful PoW. Nodes are located at AWS and other data centers.

Here is logs from one of node: https://pastebin.com/MnTeiHAM

How to resolve it?

**Comment by AmritKumar** [comment]
@bb111189 Can you comment on this?

**Comment by bb111189** [comment]
Blacklisting is a safety mechanism. If your node is unable to reach the specific node due to various network issues such as connection time out or node is unreachable, it will add it to the blacklist. 

Blacklist is local to the node and not to the whole network. It is to ensure a node won't be wasting resources sending to the unresponsive node. In this case, your node will temporary not send any message to this blacklisted nodes. Blacklist will be gradually clear once every DS epoch.

**Comment by lehisnoe** [comment]
@bb111189, I understand how is work safety mechanism. Network's issue can't occurs only for ~20-30% of nodes, located on the same servers, but uses differs IPs.
When nodes are located at AWS this problem occurs seldom.
But it happens too often when nodes starts not at AWS. So node's decentralization becomes a problem (we've some hundreeds nodes).
I'm ready to provide any detailed info via private communication. How can I communicate with you?

**Comment by bb111189** [comment]
I am re-opening this. I did probe a few of the blacklisted node in the log. It seems there are no longer running Zilliqa. 

But still, on encouraging decentralization for miners, it is something we think of in term of strategy.

**Comment by lehisnoe** [comment]
This case is not 100% reproducible - it is hard to detect this case, but if we do nothing then it stays how it is now.

**Comment by evesnow91** [comment]
@Gnnng @ansnunez Please look into this issue.

**Comment by ansnunez** [comment]
Hi @lehisnoe I analyzed your log and was able to make these findings:

1. There are 18 blacklisted IPs due to connection timeout and 2 due to connection refusal.
2. All 20 IP addresses belonged to community-owned shard nodes that existed in the network at that time.  None belonged to any Directory Service node.
3. I checked one of the Zilliqa-owned Directory Service nodes that was processing PoW submissions at that time, and it did in fact successfully receive your node's PoW submission (based on the public key in your log).
4. However, at that epoch, there were only 1695 submissions, meaning there wasn't enough to form a third shard.  As a consequence, only 1200 entries were assigned to two shards, and the rest (yours included) were not sharded in this epoch.

In short, I think this is not a blacklisting issue, but a consequence of the lack of submissions at that point in time to form the third shard.  We'll think of ways to avoid this situation by optimizing our sharding algorithm (e.g., form 3 500+ node shards instead).

**Comment by lehisnoe** [comment]
This issue is outdated. Let's close it.
